accio-tests/src/test/java/io/accio/testing/bigquery/TestRefreshCache.java

```diff
@@ -59,18 +65,24 @@ public void testRefreshFrequently()
     }
 
     private TaskInfo getTaskInfoUntilDifferent(TaskInfo taskInfo)
-            throws InterruptedException
+            throws InterruptedException, ExecutionException, TimeoutException
     {
-        TaskInfo refreshed = null;
-        while (refreshed == null ||
-                refreshed.getEndTime() == null ||
-                refreshed.getEndTime() == taskInfo.getEndTime()) {
-            refreshed = cacheManager.get().listTaskInfo(taskInfo.getCatalogName(), taskInfo.getSchemaName()).join().stream()
-                    .filter(t -> t.getTableName().equals(taskInfo.getTableName()))
-                    .findAny().orElseThrow(AssertionError::new);
-            Thread.sleep(100);
-        }
-        return refreshed;
+        return supplyAsync(() -> {
```

Comment:
Async is not required here. Could we use a timer ? And calculate `end-start` > `elapsed time`.
And if you think keep a long sleep here is required, maybe shorten it to `Thread.sleep(5000)` as we refresh cache every 5s. Add a comment like `// wait 5s for cache refresh`.

Reply:
The sleep shouldn't be equals to the refresh time. It must be less than the refresh time. If they're same, we could check the status when the refreshing is triggered every time.

Reply:
Sorry about making the confuse... I mean Thread.sleep(6000) in Line 61 , not the sleep in the while loop.

Reply:
BTW, MILLISECONDS.sleep(100) in while loop is reasonable for me.

Reply:
oh.. seems the sleeping in L61 is unnecessary. Let's remove it.